service-registry/src/main/java/io/servicecomb/serviceregistry/cache/InstanceCacheManagerOld.java

```diff
@@ -105,6 +106,30 @@ public InstanceCache createInstanceCache(String appId, String microserviceName,
     return instCache;
   }
 
+  public InstanceCache createInstanceCacheNew(InstanceCache instanceCache) {
+    FindInstancesResponse response = serviceRegistry.findServiceInstance(instanceCache);
+    List<MicroserviceInstance> instances = response.getInstances();
+    
+    if (304 == response.getStatusCode() || (Integer.valueOf(instanceCache.getRev()) - Integer.valueOf(response.getRev())) >= 0) {
```

Comment:
not only InstanceCacheManagerOld, but also InstanceCacheManagerNew need this logic
that's why i said not good bind to InstanceCache.